### DISCUSSION TITLE
"Ignore 'Failed' status" MP-wide toggle for tournaments

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#24806] [Ideas]
Currently in osu!stable, plays that are set with ScoreV2 don't submit to the map's leaderboard. This changes with osu!lazer which is great, but NoFail is almost always used in tournaments for a multitude of reasons. I don't think that the ideal solution would be to keep having tourney players use NoFail, because while those plays do submit, it doesn't feel good to have them handicapped with a 0.5x score multiplier when a lot of those plays are passes or FCs.

I think that an "Ignore 'Failed' status" MP-wide toggle would be the most ideal solution to this problem if it did the following things:

1. In the MP link, the 'Failed' status should not show up at all so that every play set in the lobby is taken into account, even if the player failed.
2. 'Failed' scores shouldn't go to the bottom of the scoreboard or the MP link below passed scores.
3. When a player's HP reaches 0 and they 'fail', there should not be a popup in the middle of the screen or any sort of greyscale dimming of the notes.
4. Team fails are not possible when using Team VS.
5. If a player fails at any point during the map, the play should not submit to the map's leaderboard.
6. If a player passes the entire map, the play should submit to the map's leaderboard.

I think that if tournaments had access to a toggle like this, it would remove the need to force NoFail for tournaments. Players who pass or FC a ranked map in a tourney would then have the play submit to the map's leaderboard without the 0.5x score multiplier associated with NoFail.

**Comment by peppy** [comment]
I think this would be very bad UX. I also stand behind the fact that if tournament players think they can pass a beatmap, they should not apply no-fail.

This is also not possible from an infrastructure perspective. In multiplayer we only submit *one* score. This is used for display in the multiplayer game but also on other leaderboards. This was an intentional choice.

Possible directions we can take this:
- Continue how it is now. Most tournaments have NoFail as a free mod, so it's up to the individual whether they apply it.
- Add a "don't submit scores to global leaderboards" aka "unranked" toggle to multiplayer rooms, so that when no fail is applied it isn't shown anywhere but in the game. This of course means that the user scores will not be included in global leaderboards.

I prefer option 1.

**Reply by Rekunan** [reply]
> Most tournaments have NoFail as a free mod, so it's up to the individual whether they apply it.

This is actually false, almost every tournament has nofail forced added upon the original mod/map so that we don't have to worry about failed scores, ever since scorev2 gave nofail a 1.0x multiplier. The exception is during freemod and tiebreakers, but everyone puts it on anyway cause, again, not having to worry about failed scores.

I'll also add that the second direction should be at least partially considered so that practice lobbies don't have their scores public.

**Reply by Flamiii** [reply]
I do agree that the UX for this kind of toggle could definitely be better, but I also think that this sort of toggle doesn't necessarily need to have the best UX because this is specifically catered towards tournament organizers and referees more than anything. It could also make sense to just have this as an MP command as well.

For the most part, tourney players do not have a choice when it comes to selecting NoFail. It's usually forced on all players to avoid things such as team fails and the MP link disregarding failed scores. And with the fact that NoFail in osu!lazer has a 0.5x score multiplier, it wouldn't work as a Freemod because it halves your score unlike in osu!stable where it has a 1x score multiplier with ScoreV2. I think that it would be a bit jarring to leave things as they are because that would make every tourney score out of 500k instead of 1 million.

In terms of the score submission that I proposed where pass scores get submitted and failed ones don't, this behaviour is identical to how osu!stable MP lobbies work already. What I am proposing is more a toggle to remove the annoying mechanics regarding failed scores in the context of an MP lobby. This means a toggle to get rid of team fails and the annoying greyscale dim and popup when your HP reaches 0, as well as the ability to remove how the MP link and scoreboard disregard failed scores.

The best 1:1 example I can provide of why this is an issue to begin with would probably be Hippo Cup 6. It was a 4v4 Team VS tournament that used ScoreV1 instead of ScoreV2, so scores did submit to the leaderboard. However with ScoreV1, NoFail has a 0.5x score multiplier, similar to how osu!lazer is now. Because it was Team VS, at least one person on each team had to take NoFail to prevent team fails, but because it had a 0.5x score multiplier, everyone else in the lobby also had to take NoFail otherwise the scores would have been extremely unbalanced. One of the scores that was set during this tournament was on [this map](https://osu.ppy.sh/beatmapsets/1020600#osu/2135493), where worst hr player got nearly 19 million score with NoFail and his score was a pass. However, instead of getting rank 1 on the map, his score was halved because of NoFail so he only got rank 8. This is going to be commonplace in osu!lazer tournaments and I just feel that it would be a missed opportunity to leave things as they are.
